AlphaWallet/KeyManagement/EtherKeystore.swift

```diff
@@ -0,0 +1,647 @@
+        keychain.delete("\(Keys.ethereumSeedUserPresenceNotRequiredPrefix)\(account.address.eip55String)")
+        keychain.delete("\(Keys.ethereumSeedUserPresenceRequiredPrefix)\(account.address.eip55String)")
+    }
+
+    private func removeAccountFromBookkeeping(_ account: EthereumAccount) {
+        ethereumAddressesWithPrivateKeys = ethereumAddressesWithPrivateKeys.filter { $0 != account.address.eip55String }
+        ethereumAddressesWithSeed = ethereumAddressesWithSeed.filter { $0 != account.address.eip55String }
+        ethereumAddressesProtectedByUserPresence = ethereumAddressesProtectedByUserPresence.filter { $0 != account.address.eip55String }
+        watchAddresses = watchAddresses.filter { $0 != account.address.eip55String }
+    }
+
+    func delete(wallet: Wallet, completion: @escaping (Result<Void, KeystoreError>) -> Void) {
+        DispatchQueue.global(qos: .userInitiated).async { [weak self] in
+            guard let strongSelf = self else {
+                return
+            }
+            let result = strongSelf.delete(wallet: wallet)
+            DispatchQueue.main.async {
+                completion(result)
+            }
+        }
+    }
+
+    func isHdWallet(account: EthereumAccount) -> Bool {
+        return ethereumAddressesWithSeed.contains(account.address.eip55String)
+    }
+
+    func isHdWallet(wallet: Wallet) -> Bool {
+        switch wallet.type {
+        case .real(let account):
+            return ethereumAddressesWithSeed.contains(account.address.eip55String)
+        case .watch:
+            return false
+        }
+    }
+
+    func isKeystore(wallet: Wallet) -> Bool {
+        switch wallet.type {
+        case .real(let account):
+            return ethereumAddressesWithPrivateKeys.contains(account.address.eip55String)
+        case .watch:
+            return false
+        }
+    }
+
+    func isWatched(wallet: Wallet) -> Bool {
+        switch wallet.type {
+        case .real:
+            return false
+        case .watch(let address):
+            return watchAddresses.contains(address.eip55String)
+        }
+    }
+
+    func isProtectedByUserPresence(account: EthereumAccount) -> Bool {
+        return ethereumAddressesProtectedByUserPresence.contains(account.address.eip55String)
+    }
+
+    func signPersonalMessage(_ message: Data, for account: EthereumAccount) -> Result<Data, KeystoreError> {
+        let prefix = "\u{19}Ethereum Signed Message:\n\(message.count)".data(using: .utf8)!
+        return signMessage(prefix + message, for: account)
+    }
+
+    func signHash(_ hash: Data, for account: EthereumAccount) -> Result<Data, KeystoreError> {
+        guard let key = getPrivateKeyForSigning(forAccount: account) else { return .failure(.accountMayNeedImportingAgainOrEnablePasscode) }
+        do {
+            var data = try EthereumSigner().sign(hash: hash, withPrivateKey: key)
+            // TODO: Make it configurable, instead of overriding last byte.
```

Comment:
Fix TODO.

Reply:
It's a leftover from old code before the PR. I think we'll always have this hardcoded for Ethereum signatures. I'll fix it outside of the PR.